active/0000-control-transpilation.md

```diff
@@ -0,0 +1,99 @@
+
+What is less know is that this configuration only affects the application code. The transpilation
+of code living in addons is done according to the configuration of those addons. While there is
+use-cases for this, particularly addons that want to run their code though a custom babel plugin,
+generally speaking the desired transpiled output is something that the end-users should control.
+
+By example, applications that only target evergreen browsers may want to disable transpilation of
+all ES6 feature since moden browsers already ship them.
+
+This RFC wants to change this behaviour so addons by default honor the host app's preferences,
+while allowing those addons to run their own transformations.
+
+# Detailed design
+
+The idea is to still allow addons to be in control of their own transpilation process, but
+make them honour uses preferenced by default.
+
+There is three different scenarios for addons:
+
+#### The addon doesn't care about the transpilation process.
+This is the most common common situation. Most addons are authored in ES6 and they don't
+mess with the host app's configuration.
+Those addons should just transpile their code user the host's apps settings.
+
+#### Addons that require experimental features
+There is two options here. Either addons can unconditionally enable that feature only for their
+tree, or it's the user responsability to enable that feature in order to use the addon, with
+optionally the addon doing some feature detection to warn (not force) the user to enable that feature.
+
+By example, an addon that uses `async-await` which right now are only available in Chrome Canary
+could either use the host app's configuration plus this feature or it could just assume the user
+has enabled it and tell them to do so in the docs.
+
+#### Addons that add their custom babel plugins
+
+The number of addons that does this is even smaller, but one of them is nothing less than Ember Data,
+which uses a babel-plugin to strip Heimdall's instrumentation from production builds.
+
+Since addons remain in control of their transpilation and just reuse the app's configuration, addons
+can attach their own plugins.
+
+#### How can addons be in charge of their own transpilation and yet honor user preferences?
+
+Ember-cli should provide to each addon a deep clone of the configuration defined in the host app
+(by example available in `this.options.babel`), and that config is the one used to transpile the addon.
+
+The addon will therefore honor the app's configuration by default, but it can also decide to add
+custom plugins, enable specific features or just throw a warning message if the app's config
+doesn't have required specific transformation enabled.
+
+By modifying this clone, their changes will not affect the host app or other addons.
+
+# How We Teach This
+
+This has no impact for apps, just for addons.
+
+Most addons will not require any change. Only those that want to modify this the transpilation
```

Comment:
I do not actually see how ember-computed-decorators needs to be changed **at all** to support this.  If you read through [the recommended snippet](https://github.com/rwjblue/ember-computed-decorators#setup-with-addon) here:

```js
  init: function(app) {
    this._super.init && this._super.init.apply(this, arguments);

    this.options = this.options || {};
    this.options.babel = this.options.babel || {};
    this.options.babel.optional = this.options.babel.optional || [];

    if (this.options.babel.optional.indexOf('es7.decorators') === -1) {
      this.options.babel.optional.push('es7.decorators');
    }
  }
```

You will notice that it already honors any `this.options.babel` settings that are present.